src/main/scala/com/lightbend/conductr/sbt/BundlePlugin.scala

```diff
@@ -203,6 +197,40 @@ object BundlePlugin extends AutoPlugin {
     )
   }
 
+  private def stageBundle(config: Configuration, bundleTypeConfig: Configuration): Def.Initialize[Task[File]] = Def.task {
+    validateSettings(config).value
+
+    val bundleTarget = (NativePackagerKeys.stagingDirectory in config).value / config.name
+    writeConfig(bundleTarget, (bundleConf in config).value)
+    val componentTarget = bundleTarget / (normalizedName in config).value
+    IO.copy((mappings in bundleTypeConfig).value.map(p => (p._1, componentTarget / p._2)))
+    componentTarget
+  }
+
+  private def stageConfiguration(config: Configuration): Def.Initialize[Task[File]] = Def.task {
+    validateSettings(config).value
+
+    val configurationTarget = (NativePackagerKeys.stagingDirectory in config).value / config.name
+    val generatedConf = (bundleConf in config).value
+    val srcDir = (sourceDirectory in config).value / (configurationName in config).value
+    if (generatedConf.isEmpty && !srcDir.exists())
+      sys.error(
+        s"""Directory $srcDir does not exist.
+            | Specify the desired configuration directory name
+            |  with the 'configurationName' setting given that it is not "default"""".stripMargin
+      )
+    IO.createDirectory(configurationTarget)
+    if (generatedConf.nonEmpty) writeConfig(configurationTarget, generatedConf)
+    IO.copyDirectory(srcDir, configurationTarget, overwrite = true, preserveLastModified = true)
+    configurationTarget
+  }
+
+  private def writeConfig(target: File, contents: String): File = {
+    val configFile = target / "bundle.conf"
+    IO.write(configFile, contents, Utf8)
+    configFile
+  }
+
```

Comment:
These functions were at the bottom of the source file and separated from their `dist` relations. I felt that it was nicer to have them back here and accordingly grouped.